idr/attributes.py

```diff
@@ -2,9 +2,13 @@
 Helper functions for accessing the IDR from within IPython notebooks.
 """
 import pandas
-from idr import create_http_session
+import omero.clients # NOQA
 from omero.rtypes import rlist, rstring, unwrap
 from omero.sys import ParametersI
+from externalDBs import get_entrezid, get_ensembleid, ensembleid_to_genesymbol
+
+from widgets import progress
```

Comment:
This is a backend library module, widgets (which requires ipython) shouldn't be a requirement.

Reply:
Oh I unfortunately have a widgets module inside idr-py, can rename it to something else if its confusing/conflicting with ipywidgets.widgets.

Reply:
[`idr/widgets.py`](https://github.com/IDR/idr-py/pull/3/files#diff-30e4a482cced41e2c7876d00b99fecb0R2) depends on ipywidgets and ipython, which means `import idr` also requires both.

Reply:
Ahhh that way, so this was my primary issue with the visualisation modules being dependent on iphython and ipywidgets. For the time being keeping the widgets within this repo as well, but can discuss more on where these modules should live.